perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -187,32 +187,53 @@ class SizeDistributionIterator(object):
     dist: dict. The distribution to draw sizes from.
   """
 
-  # IMPLEMENTATION LIMITATION: right now, this object only supports
-  # point distributions, which contain only a single size.
-
   def __init__(self, dist):
-    if len(dist) != 1:
-      raise NotImplementedError(
-          'General distributions are not implemented yet.')
-
-    size, _ = dist.iteritems().next()
-    self.size = size
+    # This class chooses random numbers by using random.random() to
+    # pick a float in [0.0, 100.0), then converting that into a choice
+    # from our distribution. The data structure is two lists, sizes
+    # and cutoffs. If the random number is in the range [cutoffs[i-1],
+    # cutoffs[i]), then the random choice is sizes[i]. Cutoffs has an
+    # implied starting element 0.0, and its last element must be
+    # 100.0.
+    self.sizes = []
+    self.cutoffs = []
+
+    total_percent = 0.0
+    for size, percent in dist.iteritems():
```

Comment:
Order matters here. You should use an `OrderedDict()`.

`dict()` does not guarantee the order of the keys. You could have the case that `next()` returns the incorrect size

Example:
```python
{1024: 10.0, 2048: 45.0, 1048576: 45.0}
```
Then but when iterating over it might give you the following sequence
```python
[(2048, 45.0), (1024, 10.0), (1048576, 45.0)]
```
and now your cutoff list is
```python
[45.0, 55.0, 100.0]
```
Now, say `random.random() * 100.0` returns `9.0`

Then your `next()` function will return `2048` but it should have been `1024`.